pkgs/os-specific/linux/system76-oled/default.nix

```
@@ -0,0 +1,35 @@
+{ rustPlatform, lib, fetchFromGitHub, pkg-config, makeWrapper, dbus, libX11
+, libXrandr }:
+rustPlatform.buildRustPackage rec {
+  pname = "system76-oled";
+  version = "0.1.4";
+
+  src = fetchFromGitHub {
+    owner = "pop-os";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256:0xy6fpr9k5wnq074flsxm1k27svvk4pvqyryhdsw1jpch46lzm1f";
+  };
+
+  nativeBuildInputs = [ pkg-config makeWrapper ];
+
+  buildInputs = [ dbus libX11 libXrandr ];
+
+  cargoBuildFlags = [ ];
+
+  cargoSha256 = "sha256:13nhs17gzya1by5b301hl8q0pm9m9k1pnzmmgx1v1qam0zd7qj91";
+
+  installPhase = ''
+    install -D -m -0644 data/system76-oled.desktop $out/etc/xdg/autostart/system76-oled.desktop
+    install -D -m 0755 target/*/release/system76-oled $out/bin/system76-oled
+  '';
```

```suggestion
  postInstall = ''
    install -D -m -0644 data/system76-oled.desktop $out/etc/xdg/autostart/system76-oled.desktop
  '';
```
The normal rust hooks already take care of copying the binary for you

---

For whatever reason, it does not appear to be the case:
```
[nix-shell:~/nixpkgs]$ nix-build -KA system76-oled
/nix/store/vqmf7b1j319pbjpnad0gzyy80c99b75g-system76-oled-0.1.4

[nix-shell:~/nixpkgs]$ ls -R result
result:
etc

result/etc:
xdg

result/etc/xdg:
autostart

result/etc/xdg/autostart:
system76-oled.desktop

[nix-shell:~/nixpkgs]$ 
```